src/wp-includes/block-editor.php

```diff
@@ -665,6 +665,8 @@ function get_block_editor_settings( array $custom_settings, $block_editor_contex
 		}
 	}
 
+	$editor_settings['canUpdateBlockBindings'] = current_user_can( 'manage_options' );
```

Comment:
I just select one of the admin capabilities according to this docs: https://wordpress.org/documentation/article/roles-and-capabilities/. `manage_options` seemed the most similar use case. But maybe it can be checked other way, maybe using `wp_get_current_user()->roles`. Not sure if that'd be better.

Reply:
I [asked on WordPress Slack](https://wordpress.slack.com/archives/C015GUFFC00/p1724923995020149?thread_ts=1724766564.036509&channel=C015GUFFC00&message_ts=1724923995.020149) whether this would work.

Reply:
1) Please don't check against a role. The roles can differ between sites and also aren't always 100% equal in a site.  For example, some sites add a role that could be thought of as in between editor and admin that might be able to manage options but not users. Some sites give contributors the ability to upload media.  I've also seen sites that rename roles to fit their organization structure. Capabilities are always the correct thing to check. 

2) Why does this need to be a higher capability than edit_post for that specific post? While I absolutely see the benefit of a separate cap and I think a `manage_bindings` that maps to `edit_post` with the specific post_id included would allow the locking down of it that is necessary for some sites.

Reply:
> While I absolutely see the benefit of a separate cap and I think a manage_bindings that maps to edit_post with the specific post_id included would allow the locking down of it that is necessary for some sites.

The biggest challenge is with site editor settings as on the client, the user might interact with many templates, template parts, etc, so we would have to check that capability through REST API somehow. I don't know too much about that are of WordPress, but happy to discuss possibilities.

Reply:
For editing existing posts, we would have a list of template/template parts and check those at load time, so it's only new ones that are added that would need to be checked by the client and I assume you worry about doing that with the REST API due to performance and if there is a wait, it's going to hurt the UX?  

What if there was a `manage_bindings` cap that mapped to the individual `edit_post` and something like `manage_all_bindings` that mapped to `edit_others_posts`?  If a user has `manage_all_bindings` then the assumption is made that they can edit it. A check can then be made with the specific ID and there will need to be some UX to say "sorry, but you actually can't do that" which might be better since it would be rare.

Reply:
I have pushed [this commit](https://github.com/WordPress/wordpress-develop/pull/7258/commits/7053c92340b8c2e1e62c67867bc6d0b219d799ce) to see how mapping a capability could look like. I mapped it to `manage_options`, but I can change it to whatever we decide.

I believe there are these options:
1. **Map `manage_block_bindings` to admin capability like `manage_options` and use that to define the `canUpdateBlockBindings` editor setting**: This would allow only admin users by default managing bindings, but admin caps don't seem too related.
2. **Map `manage_block_bindings` to editor capability like `edit_others_posts` and use that to define the editor setting**: This would allow an editor to directly manage bindings. It seems it isn't clear yet if this should be the case, though.
3. **Map `manage_block_bindings` to editor capability like `edit_others_posts`, but set `canUpdateBlockBindings` to false by default, and add a preference in the editor to enable/disable it**: This would allow us to keep the bindings UI disabled by default unless an admin explicitly sets the preference.

For all the use cases, site owners can always change the behavior by modifying the capability or the editor setting:

**Modifying the capability**

```PHP
function restrict_manage_block_bindings_cap( $caps, $cap ) {
	if ( 'manage_block_bindings' === $cap ) {
		return array( 'do_not_allow' );
	}
	return $caps;
}

add_filter( 'map_meta_cap', 'restrict_manage_block_bindings_cap', 10, 4 );
```

**Modifying the editor setting**

```PHP
function restrict_editor_setting( $editor_settings ) {
	$editor_settings['canUpdateBlockBindings'] = false;
	return $editor_settings;
}

add_filter( 'block_editor_settings_all', 'restrict_editor_setting', 10 );
```

I'd love to know your thoughts on the different possibilities.

Reply:
Option (2) is what I'd expect.

Restricting to admins as a way of slowly rolling out the feature doesn't really make sense to me. Editors should be able to do everything that an admin can do bar manage other users. 

If we want to slow down the roll out of the feature so that we have more time for feedback and testing then we should do that in more traditional ways: an experimental flag, keeping the feature only in the Gutenberg plugin, requiring the theme to opt-in, etc.

Reply:
+1 for approaching it through capabilities. This is not that different from toggling meta fields on and creating new ones in the classic editor context.

This is aside from any work to improve the interface should it not test well with users.

Reply:
>  This is not that different from toggling meta fields on and creating new ones in the classic editor context.

In that case, the `edit_post_meta` capability is the right one.  

That said, I think @noisysocks is correct and if the UI/UX needs refinement, hiding it behind a capability check isn't the right away to encourage it.

Reply:
> In that case, the edit_post_meta capability is the right one

I am not sure about this. Although post meta is the most relevant case for block bindings, the idea is that block attributes can be connected to other sources, not just post meta. Additionally, I believe connecting to post meta and editing post meta are different things.

For those reasons, I was suggesting using something more general as `edit_posts` or `edit_other_posts`.